### PR TITLE
feat: add secrets manager client factory

### PR DESCRIPTION
**Pull request by sdelamo**
see: https://github.com/micronaut-projects/micronaut-aws/issues/1006

see: https://aws.amazon.com/secrets-manager/

### CODE REVIEW

File: aws-sdk-v2/build.gradle [hunk 1]
```diff
@@ -24,6 +25,7 @@ dependencies {
     testImplementation "software.amazon.awssdk:s3"
     testImplementation "software.amazon.awssdk:dynamodb"
     testImplementation "software.amazon.awssdk:ses"
+    testImplementation 'software.amazon.awssdk:secretsmanager'
```
Comment (pgressa) [inline aws-sdk-v2/build.gradle:28]: to stay consistent you should use double quotes  here instead of simple quotes

Comment (sdelamo) [reply aws-sdk-v2/build.gradle]: Done

**Review by pgressa — APPROVED** [review]
code looks ok, just add docs, as we have for every client quite generic doc page

**Comment by pgressa** [comment]
@sdelamo maybe of this PR we could discuss this one https://github.com/micronaut-projects/micronaut-aws/pull/1017 that generates those factories instead of writing them manually.

**Comment by sdelamo** [comment]
> @sdelamo maybe of this PR we could discuss this one #1017 that generates those factories instead of writing them manually.

@pgressa  I am going to merge this PR for now. I want to develop further the integration with secret manager for distributed configuration similar AWS Parameter store and it will need to be `@BootstrapContextCompatible`.

**Comment by athkalia** [comment]
Hey @sdelamo , I am looking to load some DB credentials from AWS secrets manager to use in my lambda function that is built with Micronaut. I noticed that the guide [here](https://docs.micronaut.io/latest/guide/index.html#distributedConfigurationAwsParameterStore) only talks about the AWS parameters store. This PR seems close to what I need. Do you have any insights on whether this is possible at the moment and if so any guide for achieving what I am after? Thanks in advance! 😃

**Comment by sdelamo** [comment]
>  Do you have any insights on whether this is possible at the moment and if so any guide for achieving what I am after? Thanks in advance! 😃

It is currently not possible but I want to add that capability in the short term. We have an issue for it. https://github.com/micronaut-projects/micronaut-aws/issues/1006 Please upvote it if you are interested.

**Comment by athkalia** [comment]
Awesome! I will work around this for the time being, but I am definitely interested in trying it out once it's out! Make sure you include some docs in there when you are done 😄 🙏 

Do you have any sort of timelines in your mind by any chance?

**Pull request merged**